syntax = "proto3";

package TestMsg;

import "TestMsgDefine.proto";

message CommonInfo
{
    float  test_float  = 1;  // float  - 单精度浮点数
    double test_double = 2;  // double - 双精度浮点数
    bytes  test_string = 3;  // bytes  - 可能包含任意顺序的字节数据，类似Java的ByteString以及C++的string
    int32  test_int32  = 4;  // int32  - 使用可变长编码，编码负数不够高效，如果有负数那么使用sint32
    sint32 test_sint32 = 5;  // sint32 - 使用可变长编码，比通常的int32高效
    uint32 test_uint32 = 6;  // uint32 - 无符号整数，使用可变长编码
    int64  test_int64  = 7;  // int64  - 使用可变长编码，编码负数不够高效，如果有负数那么使用sint64
    sint64 test_sint64 = 8;  // sint64 - 使用可变长编码，比通常的int64高效
    uint64 test_uint64 = 9;  // uint64 - 无符号整数，使用可变长编码
}

message CommonInfoList
{
    repeated CommonInfo common_info_list = 1;
    repeated CommonType common_type_list = 2;
}